src/pretix/presale/views/order.py

```diff
@@ -175,23 +173,42 @@ class TicketPageMixin:
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
 
+        positions_with_tickets = self.order.positions_with_tickets
+
         ctx['order'] = self.order
 
-        can_download = all([r for rr, r in allow_ticket_download.send(self.request.event, order=self.order)])
+        can_download = bool(positions_with_tickets)
         ctx['plugins_allow_ticket_download'] = can_download
         if self.request.event.settings.ticket_download_date:
             ctx['ticket_download_date'] = self.order.ticket_download_date
-        can_download = (
-            can_download and self.order.ticket_download_available and
-            list(self.order.positions_with_tickets)
-        )
+        can_download = can_download and self.order.ticket_download_available
         ctx['download_email_required'] = can_download and (
             self.request.event.settings.ticket_download_require_validated_email and
             self.order.sales_channel == 'web' and
             not self.order.email_known_to_work
         )
         ctx['can_download'] = can_download and not ctx['download_email_required']
 
+        qs = self.context_query_set
+        if self.request.event.settings.show_checkin_number_user:
+            qs = qs.annotate(
+                checkin_count=Subquery(
+                    Checkin.objects.filter(
+                        successful=True,
+                        type=Checkin.TYPE_ENTRY,
+                        position_id=OuterRef('pk'),
+                        list__consider_tickets_used=True,
+                    ).order_by().values('position').annotate(c=Count('*')).values('c')
+                )
+            )
+        ctx['cart'] = self.get_cart(
+            answers=True, downloads=ctx['can_download'],
+            queryset=qs,
+            order=self.order
+        )
+
+        ctx['tickets_with_download'] = [p for p in ctx['cart']['positions'] if p in positions_with_tickets]
```

Comment:
Couldn’t this just be `positions_with_tickets`? Or can `positions_with_tickets` contain order-positions, that are not in cart? This loop basically just brings the `positions_with_tickets` in the correct order needed for the cart, but currently `tickets_with_download` is AFAICT only used in if-checks and the ordering there is not necessary. So:

```suggestion
        ctx['tickets_with_download'] = positions_with_tickets
```

Reply:
`positions_with_tickets` contains all positions (with tickets) from the order, on the regular `OrderDetails` page this should be a subset of `ctx['cart']['positions']`, but on the `OrderPositionDetails` page, this contains only the current position (and its addons, if any). For the `fragment_downloads.html` fragment to display the buttons properly it's necessary that only the relevant tickets are stored in `tickets_with_download`.

Reply:
Thanks for clarifying!